sbin/pfmon

```diff
@@ -172,130 +174,11 @@ exit(0);
-        }
-    );
-
-    register_task(
-        'radius audit log cleanup',
-        'radius_audit_log_cleanup_interval',
-        sub {
-            radius_audit_log_cleanup( $Config{'maintenance'}{'radius_audit_log_cleanup_window'}, $Config{maintenance}{radius_audit_log_cleanup_batch}, $Config{maintenance}{radius_audit_log_cleanup_timeout}) if ( $Config{'maintenance'}{'radius_audit_log_cleanup_window'} );
-        }
-    );
-
-    register_task(
-        'node cleanup',
-        'node_cleanup_interval',
-        sub {
-            node_cleanup( $Config{'maintenance'}{'node_cleanup_window'} ) if ( $Config{'maintenance'}{'node_cleanup_window'} );
-        }
-    );
-
-    register_task(
-        'person cleanup',
-        'person_cleanup_interval',
-        sub {
-            person_cleanup() if ( isenabled($Config{'maintenance'}{'person_cleanup'}) );
-        }
-    );
-
-    register_task(
-        'traplog cleanup',
-        'traplog_cleanup_interval',
-        sub {
-            traplog_cleanup( $Config{'maintenance'}{'traplog_cleanup_window'} ) if ( $Config{'maintenance'}{'traplog_cleanup_window'} );
-        }
-    );
-
-    register_task(
-        'checking registered nodes for expiration',
-        'nodes_maintenance_interval',
-        sub {
-            nodes_maintenance();
-        }
-    );
-
-    register_task(
-        'checking violations for expiration',
-        'violation_maintenance_interval',
-        sub {
-            violation_maintenance( $Config{maintenance}{violation_maintenance_batch}, $Config{maintenance}{violation_maintenance_timeout} );
-        }
-    );
-
-    register_task(
-        'checking accounting data for potential bandwidth abuse',
-        'inline_accounting_maintenance_interval',
-        sub {
-            inline_accounting_maintenance( $Config{'inline'}{'layer3_accounting_session_timeout'} ) if isenabled($Config{'inline'}{'accounting'});
-        }
-    );
-
-    register_task(
-        'provisioner polling enforcement',
-        'provisioning_compliance_poll_interval',
-        sub {
-            provisioner_compliance_poll();
-        }
-    );
-
-    register_task(
-        'account maintenance',
-        'acct_maintenance_interval',
-        sub {
-            acct_maintenance();
-        }
-    );
-
-    register_task(
-        'fingerbank data update',
-        'fingerbank_data_update_interval',
-        \&update_fingerbank_data
-    );
-
-    register_task(
-        'cluster_check',
-        'cluster_check_interval',
-        \&cluster_check
-    );
-
-    register_task(
-        'option82 task',
-        'option82_query_interval',
-        sub {
-            search_switch() if isenabled($Config{'network'}{'dhcpoption82logger'});
-        }
-    );
+    for my $task_id (keys %ConfigPfmon) {
+        my $task = pf::factory::pfmon::task->new($task_id);
+        next unless isenabled($task->enabled);
```

Comment:
You should create a value like $task->is_enabled that wraps the isenabled call in it